File: app/Models/Store.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Store extends Model
{
    use HasFactory;

    const CREATED_AT = 'created_at';
    const UPDATED_AT = null;

    protected $fillable = [
        'erp_store_id',
        'area_id',
        'store_code',
        'store_name',
        'store_type',
        'owner_name',
        'phone',
        'is_active'
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'created_at' => 'datetime'
    ];

    public function address(): HasOne
    {
        return $this->hasOne(StoreAddress::class);
    }

    public function area(): BelongsTo
    {
        return $this->belongsTo(Area::class);
    }

    public function assignments(): HasMany
    {
        return $this->hasMany(StoreAssignment::class);
    }

    public function visits(): HasMany
    {
        return $this->hasMany(StoreVisit::class);
    }

    public function statusHistory(): HasMany
    {
        return $this->hasMany(StoreStatusHistory::class);
    }

    public function deals(): HasMany
    {
        return $this->hasMany(Deal::class);
    }
}

File: mobile/lib/models/store.dart
import 'store_address.dart';

class Store {
  final int id;
  final String storeCode;
  final String storeName;
  final String storeType;
  final String ownerName;
  final String phone;
  final bool isActive;
  final String? erpStoreId;
  final StoreAddress? address;

  Store({
    required this.id,
    required this.storeCode,
    required this.storeName,
    required this.storeType,
    required this.ownerName,
    required this.phone,
    required this.isActive,
    this.erpStoreId,
    this.address,
  });

  factory Store.fromJson(Map<String, dynamic> json) {
    return Store(
      id: json['id'] as int,
      storeCode: (json['store_code'] ?? '').toString(),
      storeName: (json['store_name'] ?? '').toString(),
      storeType: (json['store_type'] ?? '').toString(),
      ownerName: (json['owner_name'] ?? '').toString(),
      phone: (json['phone'] ?? '').toString(),
      isActive: (json['is_active'] ?? true) == true,
      erpStoreId: json['erp_store_id']?.toString(),
      address: json['address'] is Map<String, dynamic>
          ? StoreAddress.fromJson(json['address'] as Map<String, dynamic>)
          : null,
    );
  }
}
